--- bower_components/backbone.paginator/dist/backbone.paginator.js	2024-05-16 14:03:15
+++ bower_components/backbone.paginator/dist/backbone.paginator.with_patch.js	2024-05-16 14:00:30
@@ -102,7 +102,7 @@
 
       // Create default values if no others are specified
       queryOptions = _.defaults(queryOptions, {
-        timeout: 25000,
+        timeout: 60000,
         cache: false,
         type: 'GET',
         dataType: 'jsonp'
@@ -824,22 +824,28 @@
 
       // Create default values if no others are specified
       queryOptions = _.defaults(queryOptions, {
-        timeout: 25000,
+        timeout: 60000,
         cache: false,
         type: 'GET',
         dataType: 'jsonp',
         url: self.url
       });
+
+      // Data to be sent to the server. If the HTTP method is one that cannot have an entity body,
+      // such as GET, the data is appended to the URL. We therefore need to ensure it is still encoded to succeed.
+      var decodeURIComponentIfRequired = function (value) {
+        return queryOptions.type === 'GET' ? value : decodeURIComponent(value);
+      };
 
       // Allows the passing in of {data: {foo: 'bar'}} at request time to overwrite server_api defaults
       if( options.data ){
-        options.data = decodeURIComponent($.param(_.extend(queryAttributes,options.data)));
+        options.data = decodeURIComponentIfRequired($.param(_.extend(queryAttributes,options.data)));
       }else{
-        options.data = decodeURIComponent($.param(queryAttributes));
+        options.data = decodeURIComponentIfRequired($.param(queryAttributes));
       }
 
       queryOptions = _.extend(queryOptions, {
-        data: decodeURIComponent($.param(queryAttributes)),
+        data: decodeURIComponentIfRequired($.param(queryAttributes)),
         processData: false,
         url: _.result(queryOptions, 'url')
       }, options);
